src/main/java/io/jenkins/plugins/postgresql/PostgreSQLConnection.java

```diff
@@ -0,0 +1,95 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.plugins.postgresql;
+
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Properties;
+
+/**
+ * Provides connection to a PostgreSQL instance.
+ */
+@Restricted(NoExternalUse.class)
+public class PostgreSQLConnection {
+
+    /**
+     * Create a connection to PostgreSQL.
+     */
+    static @NonNull Connection getConnection(PostgreSQLFingerprintStorage postgreSQLFingerprintStorage)
+            throws SQLException {
+        StandardUsernamePasswordCredentials standardUsernamePasswordCredentials =
+                CredentialLookup.getCredential(postgreSQLFingerprintStorage.getCredentialsId());
+        String username = CredentialLookup.getUsernameFromCredential(standardUsernamePasswordCredentials);
+        String password = CredentialLookup.getPasswordFromCredential(standardUsernamePasswordCredentials);
+        return getConnection(
+                postgreSQLFingerprintStorage.getHost(),
+                postgreSQLFingerprintStorage.getPort(),
+                postgreSQLFingerprintStorage.getDatabaseName(),
+                username,
+                password,
+                postgreSQLFingerprintStorage.getSsl(),
+                postgreSQLFingerprintStorage.getConnectionTimeout(),
+                postgreSQLFingerprintStorage.getSocketTimeout());
+    }
+
+    static @NonNull Connection getConnection(String host, int port, String databaseName, String credentialsId,
+                                             boolean ssl, int connectionTimeout, int socketTimeout)
+            throws SQLException {
+        StandardUsernamePasswordCredentials credentials = CredentialLookup.getCredential(credentialsId);
+        String username = CredentialLookup.getUsernameFromCredential(credentials);
+        String password = CredentialLookup.getPasswordFromCredential(credentials);
+        return getConnection(host, port, databaseName, username, password, ssl, connectionTimeout, socketTimeout);
+    }
+
+    /**
+     * Create a connection to PostgreSQL.
+     */
+    static @NonNull Connection getConnection(String host, int port, String databaseName, String username, String password,
+                             boolean ssl, int connectionTimeout, int socketTimeout) throws SQLException {
+        String url = "jdbc:postgresql://" + host + ":" + port + "/" + databaseName;
+
+        Properties properties = new Properties();
+        properties.setProperty("user", username);
```

Comment:
Are there constants from jdbc lib you can reuse?

Reply:
I found them here https://docs.oracle.com/database/121/JAJDB/oracle/jdbc/OracleConnection.html#CONNECTION_PROPERTY_USER_NAME. But com.oracle.ojdbc8 is not in Maven repository. Also I don't think it is the correct dependency to add since we use postgres JDBC. I was unable to find them with a quick google search, but I'll look further and report.

Reply:
I will make my own constants till the time being.

Reply:
@afalko It seems there aren't because inside the source code for PostgreSQL JDBC they are not using any constants [[source](https://github.com/pgjdbc/pgjdbc/blob/b80ebcef33f9c3801f01e28994746bc36abb2d21/pgjdbc/src/test/java/org/postgresql/test/jdbc2/LoginTimeoutTest.java#L36)].